File: conformance/bundles/07-row-check-mismatch/report.steps.kt
@file:JvmName("ReportSteps")

// Kotlin sibling of report.steps.ts / report.steps.py / ReportSteps.java
// (bundle 07-row-check-mismatch): header-bound row step — receives the current
// row (Map keyed by header cell) as the trailing argument and returns hardcoded
// (wrong) columns, producing a cell mismatch at the trace stage.
package dev.varar.kotlin.conformance.bundle07

import dev.varar.kotlin.steps
import dev.varar.kotlin.sensor

class Ctx

val steps = steps(::Ctx) {
    sensor("I report the score and grade") { row: Map<String, String> ->
        mapOf("score" to "99", "grade" to "A")
    }
}

File: conformance/bundles/07-row-check-mismatch/report.steps.go
// Go sibling of report.steps.ts (bundle 07-row-check-mismatch).
package fixture

import "github.com/varar-dev/varar-go/varar"

func Register(s *varar.Steps) {
	// Header-bound row step: returns its computed columns; the core diffs them
	// against the row cells (rowChecks). score 99 ≠ 10 → CellMismatch.
	s.Sensor("I report the score and grade", func(state varar.Value, args []varar.Value) (*varar.Value, error) {
		return varar.Ptr(varar.MapValue(map[string]varar.Value{
			"score": varar.StrValue("99"),
			"grade": varar.StrValue("A"),
		})), nil
	})
}

func State() varar.Value {
	return varar.NullValue
}
